packages/identity-sdk/src/wagmi-sdk.ts

```diff
@@ -0,0 +1,108 @@
+import { useCallback } from "react";
+import { usePublicClient, useWalletClient } from "wagmi";
+import { Address, PublicClient } from "viem";
+import {
+  initializeIdentityContract,
+  isWhitelisted,
+  generateFVLink,
+  getIdentityExpiryData,
+} from "./viem-sdk";
+
+interface IdentitySDK {
+  checkIsWhitelisted: (account: Address) => Promise<boolean>;
+  generateFVLink: (
+    callbackUrl?: string,
+    popupMode?: boolean,
+    chainId?: number,
+  ) => Promise<string>;
+  getIdentityExpiry: (account: Address) => Promise<IdentityExpiry | undefined>;
+}
+
+interface IdentityExpiry {
+  authPeriod: bigint;
+  expiryTimestamp: bigint;
+  formattedExpiryTimestamp?: string;
+  lastAuthenticated: bigint;
+}
+
+export const useIdentitySDK = (contractAddress: Address): IdentitySDK => {
+  const publicClient = usePublicClient() as PublicClient;
+  const { data: walletClient } = useWalletClient();
+
+  const contract = walletClient
+    ? initializeIdentityContract(publicClient, contractAddress)
+    : null;
+
+  const checkIsWhitelisted = useCallback(
+    async (account: Address): Promise<boolean> => {
+      if (!publicClient || !contract) {
+        throw new Error("Public client or contract not initialized.");
+      }
+      return await isWhitelisted(contract, account);
+    },
+    [contract, publicClient],
+  );
+
+  /**
+   * Generates a Face Verification Link.
+   */
+  const generateFVLinkHook = useCallback(
```

Comment:
redundant, see my sdk simply returns the functions

Reply:
Not redunant. removing this and just returning the generateFVLink method would mean a dev has to pass down the contract + walletclient.